### DISCUSSION TITLE
Blog post about advanced Faraday middleware usage

### DISCUSSION THREAD

**Discussion by mattbrictson** [lostisland/awesome-faraday#23] [Show and tell]
Hello! 👋

I wrote an in-depth blog post about implementing various HTTP client techniques with Faraday middleware. I wonder if there is a place for it in Awesome Faraday or the official Faraday docs?

> **[Advanced techniques for calling HTTP APIs in Ruby](https://mattbrictson.com/blog/advanced-http-techniques-in-ruby)**
How to use Faraday and other gems to implement circuit breakers, gracefully handle API rate limits, pool connections for better performance, and more.

I think Faraday is an amazing gem that is sometimes under-appreciated because a lot of its power requires some non-obvious configuration, and the middleware concept is kind of abstract. I'm hoping my post makes it a little more approachable.

**Comment by olleolleolle** [comment]
@mattbrictson If you find a nice place in the Faraday documentation, perhaps in the final parts of it (when the reader has passed through the basic things), then linking to this blog post seems pertinent, and I welcome it.

Good post! Generous! Clear!

**Comment by iMacTia** [comment]
Thank you @mattbrictson, really useful blog post 👏 
It does a great job of clarifying and demystifying some common issues and pitfalls that users can face using Faraday, and it showcases how easy it is to build robust HTTP clients.
I've linked it in the main [Faraday REAME](https://github.com/lostisland/faraday#getting-started) 🙏 

Documentation is definitely an area where we're lacking and I can't wait to have some spare time to work on it 😄

**Comment by mattbrictson** [comment]
Thanks @olleolleolle @iMacTia ! 🙇
